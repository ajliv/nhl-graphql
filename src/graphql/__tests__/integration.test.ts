import { gql } from 'apollo-server-express';
import { createTestClient } from 'apollo-server-testing';

import { mockPerson, mockNHL } from '../datasources/__tests__/fixtures';

import { createGraphQLServer } from '..';

const GET_ANTOINE = gql`
    query getAntoine($id: Int) {
        player(id: $id) {
            birthDate
            fullName
            position
            primaryNumber
            currentTeam {
                name
                abbreviation
            }
        }
    }
`;

describe('Queries', () => {
    it('fetches antoine', async () => {
        const nhl = mockNHL();
        const server = createGraphQLServer({
            dataSources: () => ({ nhl }),
        });
        const { query } = createTestClient(server);
        const res = await query({
            query: GET_ANTOINE,
            variables: { id: 8474849 },
        });

        expect(res).toMatchSnapshot();
        expect(nhl.getPerson).toHaveBeenCalledWith(8474849);
        expect(nhl.getTeam).toHaveBeenCalledWith(mockPerson.currentTeam.id);
    });
});
