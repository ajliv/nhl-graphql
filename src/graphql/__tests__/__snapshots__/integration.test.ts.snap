// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Queries fetches antoine 1`] = `
Object {
  "data": Object {
    "player": Object {
      "birthDate": "1989-11-21T00:00:00.000Z",
      "currentTeam": Object {
        "abbreviation": "DAL",
        "name": "Dallas Stars",
      },
      "fullName": "Antoine Roussel",
      "position": "LW",
      "primaryNumber": 26,
    },
  },
  "errors": undefined,
  "extensions": undefined,
  "http": Object {
    "headers": Headers {
      Symbol(map): Object {},
    },
  },
}
`;
